### PR TITLE
test: fix flaky integ tests

### PR DESCRIPTION
**Pull request by bvtujo**
This PR fixes the flaky template integration tests which were a result of rendering maps to lists from manifest to template. 

The new test exercises the autoscaling configuration and empty volume logic for load balanced web services. 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: internal/pkg/stream/stream_test.go [hunk 1]
```diff
@@ -65,6 +65,10 @@ func TestStream(t *testing.T) {
 		// GIVEN
 		ctx, cancel := context.WithCancel(context.Background())
 		cancel()
+		// This test is flaky and sometimes results in a single call to fetch instead of none if there is no delay between
+		// cancel() and Stream().
+		// We wait 5 ms to ensure the cancel message is received.
+		time.Sleep(5 * time.Millisecond)
```
Comment (iamhopaul123) [inline internal/pkg/stream/stream_test.go]: nice! it makes sense to me.

**Review by iamhopaul123 — APPROVED** [review]
ship it 🚢

File: internal/pkg/stream/stream_test.go (hunk 1 above)
Comment (efekarakus) [inline internal/pkg/stream/stream_test.go]: With the given description i think it makes sense for us to remove this test case completely and just rely on the `"calls Fetch and Notify multiple times until context is canceled"` test case

File: templates/workloads/partials/cf/service-base-properties.yml [hunk 2]
```diff
@@ -7,9 +7,6 @@ DesiredCount: !GetAtt DynamicDesiredCountAction.DesiredCount
 {{- else }}
 DesiredCount: !Ref TaskCount
 {{- end}}
-{{- if .Storage}}
```
Comment (efekarakus) [inline templates/workloads/partials/cf/service-base-properties.yml:10]: Yay Thanks for this change 🥳

**Review by efekarakus — APPROVED** [review]
Looks good just a small request, and then we can remove the label!

File: internal/pkg/stream/stream_test.go (hunk 1 above)
Comment (bvtujo) [reply internal/pkg/stream/stream_test.go]: Nice, I'll just cut this test entirely.

**Pull request merged**